### PR TITLE
usb_host: add building filesystem on USB stick step

### PR DESCRIPTION
**Pull request by ybduan**
Build new filesystem on USB stick step before getting the mount
point.

ID: 2027235

Signed-off-by: ybduan <yduan@redhat.com>

### CODE REVIEW

**Comment by ybduan** [comment]
Summary: 
Finshed=18, PASS=18

**Comment by ybduan** [comment]
Please help review this PR, thanks! @yanan-fu

File: qemu/tests/usb_host.py [hunk 1]
```diff
@@ -75,14 +75,18 @@ def _get_usb_mount_point():
...
-            mp = "/mnt"
-        else:
-            mp = re.findall(r'/dev/%s\d*\s+(\S+)\s+' % dev, o)[0]
-        return mp
+                test.error("Fail to umount /dev/%s, output: %s" % (s, o))
+        mkfs_cmd = "mkfs.vfat /dev/%s" % dev
+        s, o = session.cmd_status_output(mkfs_cmd)
+        if s:
+            test.error("Fail to build filesystem on usb stick: %s" % o)
+        s, o = session.cmd_status_output("mount /dev/%s /mnt" % dev)
```
Comment (yanan-fu) [inline qemu/tests/usb_host.py:86]: IMO, it is better to make the mount point configurable.

File: qemu/tests/usb_host.py [hunk 2]
```diff
@@ -75,14 +75,18 @@ def _get_usb_mount_point():
...
+        if not s:
+            s, o = session.cmd_status_output('umount /dev/%s' % dev)
             if s:
-                test.error("Fail to mount /dev/%s, output: %s" % (s, o))
-            mp = "/mnt"
-        else:
-            mp = re.findall(r'/dev/%s\d*\s+(\S+)\s+' % dev, o)[0]
-        return mp
+                test.error("Fail to umount /dev/%s, output: %s" % (s, o))
+        mkfs_cmd = "mkfs.vfat /dev/%s" % dev
```
Comment (yanan-fu) [inline qemu/tests/usb_host.py:82]: No matter the status of usb stick, make this step as must, clean all the existing data if there are ?

Comment (ybduan) [reply qemu/tests/usb_host.py:82]: Updated, thanks!

File: qemu/tests/usb_host.py (hunk 1 above)
Comment (ybduan) [reply qemu/tests/usb_host.py]: Updated, thanks!

File: qemu/tests/usb_host.py (hunk 2 above)
Comment (ybduan) [reply qemu/tests/usb_host.py:82]: Yes! The purpose of this PR is to umount the passthrough USB stick if it's already mounted, re-make the filesystem on the USB stick to prepare a clean test environment, and then mount it (again) to guest for the following r/w testing.

File: qemu/tests/usb_host.py [hunk 3]
```diff
@@ -75,14 +75,19 @@ def _get_usb_mount_point():
...
+        s, o = session.cmd_status_output(mkfs_cmd)
+        if s:
+            test.error("Fail to build filesystem on usb stick: %s" % o)
+        mount_point = params.get("mount_point", "/mnt")
```
Comment (yanan-fu) [inline qemu/tests/usb_host.py:86]: Check if `mount_point` exist or not, create it if not.

File: qemu/tests/usb_host.py (hunk 2 above)
Comment (yanan-fu) [reply qemu/tests/usb_host.py:82]: ok, thanks!

File: qemu/tests/usb_host.py (hunk 3 above)
Comment (ybduan) [reply qemu/tests/usb_host.py:86]: Updated, thanks!

**Comment by yanan-fu** [comment]
Could you please provide test result ?  thanks

**Comment by ybduan** [comment]
RESULTS    : PASS 18 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2021-12-16T03.15-f033d7a/results.html
JOB TIME   : 64915.44 s

**Review by yanan-fu — APPROVED** [review]
Ack

**Pull request merged**